lib/src/st2110/st_video_transmitter.c

```diff
@@ -457,6 +458,94 @@ static int video_trs_tsc_tasklet(struct mtl_main_impl* impl,
   return MT_TASKLET_HAS_PENDING;
 }
 
+static int video_trs_launch_time_tasklet(struct mtl_main_impl* impl,
+                                         struct st_tx_video_session_impl* s,
+                                         enum mtl_session_port s_port) {
+  unsigned int bulk = s->bulk;
+  struct rte_ring* ring = s->ring[s_port];
+  int tx = 0;
+  unsigned int n;
+  uint64_t i;
+  uint64_t target_ptp;
+  uint16_t port_id = s->port_id[s_port];
+  struct mt_interface* inf = mt_if(impl, port_id);
+  struct mt_ptp_impl* ptp = mt_get_ptp(impl, port_id);
+
+  if (!ptp->phc2sys.stat_sync) {
```

Comment:
If time sync not ready, do nothing.

Reply:
Can you add a assignment to the trs ret code also? That we can know the trs is blocking as the ptp timesync not ready.
```
s->stat_trs_ret_code[s_port] = -STI_TSCTRS_BURST_TIMESYNC_NOT_READY;
```

And what you say is `time sync not ready`, but the code actually check is phc2sys not ready. Can you clarify?

Reply:
New implement! If time sync not ready `ptp->phc2sys.stat_sync = false`, execute the old fashion pacing.

Reply:
Sounds good. But I still not sure why it check `phc2sys.stat_sync` instead of the ptp status? Does it mean the TSN pacing can only works with both PTP and phc2sys are enabled? phc2sys is a must required for TSN pacing?

Reply:
Currently phc2sys is NOT required for TSN pacing. 

Considering that Yu, Ping said IMPL will use system clock to replace PTP clock in future. And phc2sys synced is the sufficient conditions of PTP synced. So, I used phc2sys synced for time synced indicator.

Reply:
IMTL support different ptp time source, built-in or system clock(ptp4l+phc2sys). Even for TSN, the phc2sys you added is just an option and need root access. If timesync is enough, please only check timesync status here.